src/rpc/methods/eth.rs

```diff
@@ -969,18 +969,22 @@ fn new_eth_tx_from_message_lookup<DB: Blockstore>(
     let parent_ts_cid = parent_ts.key().cid()?;
 
     // Lookup the transaction index
-    if tx_index < 0 {
-        let msgs = ctx.chain_store().messages_for_tipset(&parent_ts)?;
-        for (i, msg) in msgs.iter().enumerate() {
-            if msg.cid() == message_lookup.message {
-                tx_index = i as i64;
-                break;
+    let tx_index = {
+        if let Some(index) = tx_index {
```

Comment:
nit: you can probably do something like `tx_index.unwrap_or_else(your_fallback_code).